identity-iota-core/src/diff/diff_iota_document.rs

```diff
@@ -55,7 +55,7 @@ impl Diff for IotaDocument {
       .transpose()?
       .unwrap_or_else(|| self.metadata.clone());
 
-    Ok(IotaDocument::from((document, metadata)))
+    Ok(IotaDocument::from((document, metadata, None)))
```

Comment:
It might be worth adding the same comment that was in `DiffIotaDocumentMetadata` that the proof is intentionally excluded from the diff.

`// NOTE: proof intentionally excluded.`